opentreemap/importer/tasks.py

```diff
@@ -67,7 +67,7 @@ def _create_rows(ie, reader):
         RowModel.objects.bulk_create(rows)  # create final partial block
 
 
-@task()
+@shared_task()
 def run_import_event_validation(import_type, import_event_id, file_obj):
```

Comment:
If an import file has fewer rows than our row chunk size (85) no import rows were being created. It turns out that an exception was being thrown due to the fact that celery 4 now converts a chord with only one task in the head into a simple AsyncResult, which does not have a savable group result parent (http://docs.celeryproject.org/en/latest/whatsnew-4.0.html)

Here is a test file with one species that will work for an instance centered on LA that works in staging but not on this branch [la_test_species_import.csv.zip](https://github.com/OpenTreeMap/otm-core/files/1232322/la_test_species_import.csv.zip)

Wrapping the group result saving in a conditional fixed the issue for me

```diff
diff --git a/opentreemap/importer/tasks.py b/opentreemap/importer/tasks.py
index 2702940c..9b0c0e5f 100644
--- a/opentreemap/importer/tasks.py
+++ b/opentreemap/importer/tasks.py
@@ -6,6 +6,7 @@ from __future__ import division
 import json
 
 from celery import shared_task, chord
+from celery.result import GroupResult
 from django.core.exceptions import ObjectDoesNotExist
 from django.conf import settings
 from django.db import transaction
@@ -99,10 +100,14 @@ def run_import_event_validation(import_type, import_event_id, file_obj):
         final_task = _finalize_validation.si(import_type, import_event_id)
 
         async_result = chord(validation_tasks, final_task).delay()
-        group_result = async_result.parent
-        if group_result:  # Has value None when run in unit tests
-            group_result.save()
-            ie.task_id = group_result.id
+        async_result_parent = async_result.parent
+        if async_result_parent:  # Has value None when run in unit tests
+            if isinstance(async_result_parent, GroupResult):
+                # Celery 4 converts a chord with only one task in the head into
+                # a simple chain, which does not have a savable parent
+                # GroupResult.
+                async_result_parent.save()
+            ie.task_id = async_result_parent.id
 
         _assure_status_is_at_least_verifying(ie)
 

```